### PR TITLE
[WIP] Regen chunk meshes to remove chunk walls

### PR DESCRIPTION
**Pull request by afrigon**
This works although it regenerate the mesh data 4 to 5 times per chunk and has a O(n^2) check that slows the fps down more than simply rendering the world borders

an alternate solution would be to hold a list of chunk updates and the renderer could consume these to regenerate the chunk without a complex check. this solution would also implement the logic for updating chunk on placing and removal of blocks.

this is trying to solve #17

### CODE REVIEW

**Comment by filedesless** [comment]
blocked by roadrunner-craft/core#18

File: src/render/models/chunk_mesh.rs [hunk 1]
```diff
@@ -267,7 +267,7 @@ impl ChunkMesh {
             mesh.model = Some(Model::new(&mesh.vertices, &mesh.indices));
```
Comment (afrigon) [inline src/render/models/chunk_mesh.rs:267]: make the model not optional and return none if the mesh has no vertex

File: src/render/renderer/chunk_renderer.rs [hunk 2]
```diff
@@ -151,29 +151,16 @@ impl ChunkRenderer {
...
-            .filter(|coords| !self.meshes.contains_key(coords))
-            .collect::<Vec<&ChunkGridCoordinate>>();
-
         // remove unloaded chunk and new chunks neighbour
         self.meshes.retain(|coords, _| {
-            world.chunks.contains_key(coords)
-                && new_chunks
-                    .iter()
-                    .all(|other| !ChunkGridCoordinate::are_neighbours(coords, other))
+            world.chunks.contains_key(coords) && !world.meshes_to_regen.contains(coords)
```
Comment (afrigon) [inline src/render/renderer/chunk_renderer.rs:156]: kinda hate that name meshes_to_regen, what about chunk_updates

File: src/render/renderer/chunk_renderer.rs [hunk 3]
```diff
@@ -151,29 +151,16 @@ impl ChunkRenderer {
...
         // generate missing geometry for loaded chunks
-        for (coords, _) in world.chunks.iter() {
-            if !self.meshes.contains_key(coords) {
-                let chunk_group = world.get_chunk_group(*coords);
-                self.meshes.insert(
-                    *coords,
-                    ChunkMesh::generate(&chunk_group, &self.block_registry),
-                );
+        for coords in world.meshes_to_regen.iter() {
+            let chunk_group = world.get_chunk_group(*coords);
```
Comment (afrigon) [inline src/render/renderer/chunk_renderer.rs:161]: make the getter follow our code style chunk_group() instead of get_chunk_group

**Pull request closed without merging**